cmake/Platform/Other/FindArduinoSDK.cmake

```diff
@@ -25,6 +25,10 @@ function(find_arduino_sdk _return_var)
                 "If so, please set the ARDUINO_SDK_PATH CMake-Variable")
         message(FATAL_ERROR ${error_message})
     else ()
+        if (${CMAKE_HOST_APPLE})
+            get_filename_component(sdk_path "${sdk_path}" DIRECTORY)
```

Comment:
I'm having trouble to understand exactly what path does the `sdk_path` variable resolves to originally, before getting its' parent directory. 
I took a look at the **OS X** Arduino SDK and it seems that:
1. The entire SDK sits under `Contents/Java`
2. There's no file named **arduino** in this directory, only files that include this name partially, such as `arduino-builder`.
AFAIK, CMake's `find_program` doesn't search recursively , so how can it find the **arduino** program?
3. Let's say it does take **arduino-builder** as the program it finds, then sets `sdk_path` to **Java.**
In this case, the code you've added sets `sdk_path` to **Contents** instead (parent dir of **Java**), then appends **Java** back - WTH? This is the same path as it would be without this code addition.

As you see, I'm very confused here and unfortunately I don't have a Mac to check this on, nor I can trigger a build on 'Travis' to output this info for me right now.

Reply:
The original code will get `arduino_program_path` with `/Applications/Arduino.app/Contents/MacOS/Arduino` and `sdk_path` will be `/Applications/Arduino.app/Contents/MacOS`. The proper `sdk_path` should be `/Applications/Arduino.app/Contents/Java`.

So we can fix it with:
```
get_filename_component(sdk_path "${sdk_path}" DIRECTORY)
string(APPEND sdk_path "/Java")
```

And there is [another way](https://github.com/taoyuan/arduino-cmake/blob/master/cmake/ArduinoToolchain.cmake#L74) to get `sdk_path`. You can take a look.

Reply:
Oh I get it now, thanks!
I'll approve this PR and will integrate the ["better" way](https://github.com/taoyuan/arduino-cmake/blob/master/cmake/ArduinoToolchain.cmake#L74) later on in a separate PR.